sql/sys_vars.h

```diff
@@ -550,6 +550,53 @@ class Sys_var_charptr: public sys_var
 };
 
 
+class Sys_var_version : public Sys_var_charptr
+{
+private:
+  char withsuffix[SERVER_VERSION_LENGTH];
+  char *withsuffix_ptr;
+public:
+  Sys_var_version(const char *name_arg,
+          const char *comment, int flag_args, ptrdiff_t off, size_t size,
+          CMD_LINE getopt,
+          enum charset_enum is_os_charset_arg,
+          const char *def_val)
+    : Sys_var_charptr(name_arg, comment, flag_args, off, size, getopt,
+          is_os_charset_arg, def_val),
+      withsuffix_ptr(withsuffix)
+  {}
+  virtual uchar *global_value_ptr(THD *thd, LEX_STRING *base)
+  {
+    char **version_ptr= reinterpret_cast<char**> (
+                               Sys_var_charptr::global_value_ptr(thd, base));
+    if (version_ptr == NULL)
+      return NULL;
+
+    sys_var *suffix_var= find_sys_var(thd, STRING_WITH_LEN("version_suffix"));
+    if (suffix_var == NULL)
+      return reinterpret_cast<uchar*> (version_ptr);
+
+    char** suffix_ptr= reinterpret_cast<char**> (suffix_var->value_ptr(thd,
```

Comment:
Can there be a race condition if another thread modifies `version_suffix` at the same time?

Reply:
This commit does not change anything in update of variables. They are modified using
`sys_var::update(THD *thd, set_var *var)`
which uses
```
    AutoWLock lock1(&PLock_global_system_variables);
    AutoWLock lock2(guard);
```